pandas/io/sql.py

```diff
@@ -1232,10 +1232,15 @@ def _get_dtype(self, sqltype):
             DateTime,
             Float,
             Integer,
+            Numeric,
         )
 
         if isinstance(sqltype, Float):
             return float
+        elif isinstance(sqltype, Numeric) and sqltype.scale is None:
+            return np.dtype("int64")
+        elif isinstance(sqltype, Numeric) and sqltype.scale > 0:
```

Comment:
Wierd case. Oracle NUMBER inherits from both Numeric and Integer. An Integer will come back as a Numeric with None in scale and as an Integer, whereas a true float will come back as Numeric with Scale > 0. If I don't include that check, all oracle NUMBERS get cast to float, which while not losing precision, may be confusing for downstream use. I also tried putting it after the integer check, but led back to the issue I was trying to resolve.

Reply:
Ah OK. Is that specific to Oracle? Curious if you've checked upstream for any bug reports in SQLAlchemy regarding this behavior?

Reply:
Found this article in SQLAlchemy:

https://docs.sqlalchemy.org/en/14/dialects/oracle.html#precision-numerics

Instead of doing this generally could we instead just check if we are dealing with a `sqlalchemy.dialects.oracle.Number` type and special case that as needed? The way this is written now I think applies to all databases but this seems really specific

Reply:
Thanks for the feedback. That makes total sense so I've incorporated that change.